public/app/plugins/panel/timeseries/TimeSeriesPanel.tsx

```diff
@@ -36,7 +37,23 @@ export const TimeSeriesPanel = ({
   const { sync, canAddAnnotations, onThresholdsChange, canEditThresholds, showThresholds, dataLinkPostProcessor } =
     usePanelContext();
 
-  const frames = useMemo(() => prepareGraphableFields(data.series, config.theme2, timeRange), [data.series, timeRange]);
+  let prevTimeRange = usePrevious(timeRange);
+  let prevFrames = usePrevious(data.series);
+
+  console.log('sameTimeRange', timeRange === prevTimeRange);
+  console.log('sameFrames', data.series === prevFrames);
+  console.log('compareFrames', compareArrayValues(prevFrames ?? [], data.series, (a, b) => a === b));
+
+  const frames = useMemo(() => {
+    console.log({
+      state: data.state,
+      interval: data.series[0].fields[0].config.interval,
+      timeRange: `${Math.round((timeRange.to.valueOf() - timeRange.from.valueOf()) / 1000 / 3600)} hrs`,
+    });
```

Comment:
if `data.series` and `timeRange` props are same during `Loading` state, then this should only log once during `Done` state, and not during `Loading` state.